core/src/main/scala/com/mesosphere/package.scala

```diff
@@ -0,0 +1,103 @@
+package com
+
+import com.typesafe.scalalogging.CanLog
+import com.typesafe.scalalogging.Logger
+import com.typesafe.scalalogging.LoggerTakingImplicit
+import org.slf4j.MDC
+
+package object mesosphere {
+  implicit case object CanLogKvArgsEv extends CanLog[KvArgs] {
+    override def logMessage(originalMsg: String, a: KvArgs): String = {
+      for (elem <- a.args) {
+        val (key, value) = elem
+        MDC.put(key, value.toString)
+      }
+      originalMsg
+    }
+
+    override def afterLog(a: KvArgs): Unit = a.args.foreach { case (key, _) => MDC.remove(key) }
+  }
+}
+
+package mesosphere {
+
+  /**
+   * Abstractions in this class were designed with below interests:
+   * 1. Use the slf4j abstraction without any backend (such as logback etc.,). Consumers of this library should be
+   *    free to chose whatever backend they wish to use. `MDC` is a well-known concept in java world and many logging
+   *    backends will be able to do a structured logging that includes MDC.
+   * 2. Make it easy to compose context(s) E.g.: [[Traversable]] of tuples of type ([[String]], [[Any]]) for loggers.
+   * 3. User should be able to :
+   *     - Mix in one of the [[ImplicitStrictLogging]] or [[ImplicitLazyLogging]] traits
+   *     - Use the logger instance `logger` to print log statements with some context parameters.
+   *     - Pass the context to another method or class which has no information about context built so far
+   *       and thus can chose to built context further on top of the current context OR start fresh.
+   *
+   * Following snippet shows an example of how the [[LoggerTakingImplicit]] can be used with the
+   * abstractions defined here :
+   *
+   * {{{
+   *    // import all the implicit defined here in to scope.
+   *    import com.mesosphere._
+   *
+   *    // Mixin one of the traits.
+   *    class MyClass with ImplicitStrictLogging {....
+   *
+   *    // Make a log statement with no (empty) context.
+   *    logger.info("hello world!")(KvArgs())
```

Comment:
do you need to do it like this or will logger.info("aaa") work as well?

Reply:
as of now, we need to pass an empty `KvArgs()`. If this tends out to be a common pattern, we can bring back the global implicit for empty `KvArgs` https://github.com/mesosphere/usi/pull/36#discussion_r257237992

Reply:
brought the global implicit back :)